week3/homework/src/actions.js

```diff
@@ -0,0 +1,100 @@
+let fs = require('fs');
+
+function readTodo(request, response) {
+  fs.readFile('./todoList.json', 'utf8', (error, todoList) => {
+    const parsedList = JSON.parse(todoList);
+    if (error) {
+      if (error.code === 'EN0ENT') {
+        console.log('No data found');
+      } else {
+        console.error(error);
+        response.status(500).send('some problem.');
+      }
+    } else if (request.params.id < parsedList.length) {
+      const todoItem = parsedList[++request.params.id]['text'];
+      response.json(todoItem);
+    } else {
+      response.json('Please enter a valid id');
```

Comment:
When the user provides invalid input, the response code should be 4xx, in you case `400` which means Bad Request, also the message you are sending back is not JSON, so you use `response.send`;  So `response.status(400).send('Please enter a valid id')`

Even better if you split it into two cases;
- When ID <= 0 it means Bad Request (400)
- When ID > Todo-List-Size  it means Not Found (404)